velox/docs/develop/timestamp.rst

```diff
@@ -0,0 +1,301 @@
+outside of Velox execution, they are exposed if data containing
+TimestampWithTimezones are serialized, for example.
+
+**TimestampWithTimezone:** To represent an absolute point in time, Velox provides
+the `TimestampWithTimezone class <https://github.com/facebookincubator/velox/blob/main/velox/functions/prestosql/types/TimestampWithTimeZoneType.h>`_.
+This class implements the TIMESTAMP WITH TIMEZONE sql semantic discussed above,
+and was based on Presto’s implementation - therefore only supporting
+millisecond-precision.
+
+TimestampWithTimezone physically packs two integers in a single 64 word, using
+12 bits for timezone ID, and 52 for a millisecond-precision timestamp.
+
+Note that to accelerate timestamp conversion functions, the timestamps stored
+in a TimestampWithTimezone are **always relative to UTC** - they are unix epochs.
+This means that converting a TimestampWithTimezone across timezones is
+efficiently done by just overwriting the 12 bits, and that comparisons can be
+done by just comparing the 52 bits relative to timestamp (ignoring the timezone
+ID).
+
+However, unpacking/converting a TimestampWithTimezone into an absolute time
+definition requires a
+`timezone conversion <https://github.com/facebookincubator/velox/blob/main/velox/functions/prestosql/DateTimeFunctions.h#L74-L84>`_. 
+
+Conversions Across Timezones
+----------------------------
+
+A common operation required when processing timestamps and timezone is finding
+the wall clock and calendar read in a specific timezone given an absolute point
+in time described by a wall clock and calendar read in a different timezone.
+For example, at the exact point in time when UTC hits ``1970-01-01 00:00:00``,
+what was the wall clock read in China?
+
+Timezone conversions are tricky since they are non-linear and depend on
+daylight savings time schedules and other local regulations, and these change
+over time. To enable such conversions, `IANA <https://www.iana.org/time-zones>`_
+periodically publishes a global source of authority database for timezone
+conversions, which is periodically pushed to systems using packages like tzdata
+for Linux. 
+
+In Velox, Timezone conversions are done using std::chrono. Starting in C++20,
+std::chrono `supports conversion of timestamp across timezones <https://en.cppreference.com/w/cpp/chrono/time_zone>`_.
+To support older versions of the C++ standard, in Velox we vendor an
+implementation of this API at `velox/external/date/ <https://github.com/facebookincubator/velox/tree/main/velox/external/date>`_.
+This class handles timezone conversions by leveraging APIs provided by the
+operating system, based on the tzdata database installed locally. If systems
+happen to have inconsistent or older versions of the tzdata database, Velox’s
+conversions may produce inconsistent results. 
+
+You can check the tzdata installed in your system by:
+
+.. code-block:: bash
+
+  $ rpm -qa | grep tzdata
```

Comment:
I tried this command on my Mac, but it didn't print anything.

Reply:
this is for linux; added to the text to clarify